internal/test/assert.go

```diff
@@ -0,0 +1,62 @@
+package test
+
+import (
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+type checkerOptions struct {
+	bodyChecker func(*testing.T, []byte) bool
+}
+
+type CheckerOption func(*checkerOptions)
```

Comment:
This type is called `CheckerOption` in a package called `test`. This might not be ideal since it's a bit vague.

How about calling this `HTTPResponseOpt`? + see below suggestion.